### PR TITLE
Log-2769: Support Elasticsearch 8.x as an output using vector

### PR DESCRIPTION
**Pull request by Clee2691**
#### Summary:
- Added optional `Elasticsearch.Version` field to types.
- Added `suppress_type_name = true` if forwarding to `Elasticsearch 7.x` or higher.
- Can forward both to `default` and external `Elasticsearch` store at the same time with a `ClusterLogForwarder` custom resource.

#### Details:

This PR addresses an issue with forwarding logs using `vector` to `Elasticsearch 8.x` (ES). Forwarding logs to ES 8.x produced the following error in the logs: `Action/metadata line [1] contains an unknown parameter [_type]`.

To fix the error, in the `vector.toml` config file, `[sinks - <ES>]` section, `suppress_type_name = true` needs to be added for ES versions 7.x and higher. However, we cannot suppress the type name for ES 6.x, which is our internally managed ES.

To distinguish between ES versions, in the `output_types.go` file I added the `Version` field under `Elasticsearch`. This field is optional and will default to the latest ES version (right now that is 8.x) if not specified by the end user when defining a `ClusterLogForwarder` custom resource. The internal `default` store will be ES 6.x.

With this field, the `suppress_type_name = true` will only be added to `vector.toml` ES sink configurations using version 7.x or higher.

/cc @cahartma   

### Links

- https://issues.redhat.com/browse/LOG-2769

### CODE REVIEW

**Review by cahartma — CHANGES_REQUESTED** [review]
I've added a comment on forwarding.go

File: internal/k8shandler/forwarding.go [hunk 1]
```diff
@@ -25,8 +25,52 @@ import (
 
 func applyOutputDefaults(outputDefaults *logging.OutputDefaults, out logging.OutputSpec) logging.OutputSpec {
 	if outputDefaults != nil && outputDefaults.Elasticsearch != nil {
-		if out.Type == logging.OutputTypeElasticsearch && out.Elasticsearch == nil {
-			out.Elasticsearch = outputDefaults.Elasticsearch
+		if out.Type == logging.OutputTypeElasticsearch {
```
[the remark this answers is not in the document]
Comment (Clee2691) [reply internal/k8shandler/forwarding.go]: Refactored by encapsulating determining and setting Elasticsearch version into a separate function.

[the remark this answers is not in the document]
Comment (cahartma) [reply internal/k8shandler/forwarding.go]: Refactor looks good.  The additional comments help as well.      
We are holding for after release of 5.5

**Comment by jcantrill** [comment]
master is open for 5.6

**Review by cahartma — COMMENTED** [review]
/lgtm

**Comment by cahartma** [comment]
Rebased in PR https://github.com/openshift/cluster-logging-operator/pull/1668

**Pull request closed without merging**